app/models/admin.rb

```diff
@@ -1,9 +1,11 @@
 class Admin < ActiveRecord::Base
-  after_save :update_prefixes
+  after_save :update_active_slug_prefixes, if: -> { product_slug_changed? }
 
   private 
   
-  def update_prefixes
-    Slugs::UpdateSlugPrefixesForActiveSlugs.call(resource_type: 'product') if product_slug_changed?
+  def update_active_slug_prefixes
+    Slugs::UpdateSlugPrefixesForActiveSlugsWorker.perform_async(
+      resource_type: 'product', slug_prefix: Product.slug_prefix
```

Comment:
`resource_type` may need to be `Product` as it should be the class name, obviously when merging to Shift, this would be dynamic anyway.